### PR TITLE
Backport of the new test pick a free port strategy

### PR DESCRIPTION
**Pull request by elpaso**

### CODE REVIEW

**Comment by nyalldawson** [comment]
Great work!

**Comment by elpaso** [comment]
@nyalldawson thanks, but better wait and see if that works :roll_eyes: I've never been able to reproduce a single failure locally even with the previous implementations.

This should be bulletproof: the server picks a free port and binds, then it prints the port on `stdout` and the caller reads that value, then the caller tries to connect to the server until it times out or succeed, at that point the server should really be listening.

**Comment by elpaso** [comment]
@m-kuhn any idea about what's happening here https://travis-ci.org/qgis/QGIS/jobs/164872372 ?

File: tests/src/python/utilities.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
 import glob
 import platform
 import tempfile
+from urllib2 import urlopen, HTTPError
```
Comment (m-kuhn) [inline tests/src/python/utilities.py]: http://stackoverflow.com/questions/2792650/python3-error-import-error-no-module-name-urllib2

**Pull request merged**